src/main/java/com/altenheim/kalender/controller/logicController/DateSuggestionController.java

```diff
@@ -0,0 +1,39 @@
+package com.altenheim.kalender.controller.logicController;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import com.altenheim.kalender.interfaces.IDateSuggestionController;
+import com.calendarfx.model.Entry;
+
+public class DateSuggestionController implements IDateSuggestionController 
+{
+    public Entry<String> getDateSuggestionFromEntryList(ArrayList<Entry<?>> input, LocalDateTime startSearchDateTimeInput, int dateLenght) 
+    {
+        var startSearchDateTime  = LocalDateTime.of(startSearchDateTimeInput.toLocalDate(), startSearchDateTimeInput.toLocalTime());
+        
+        if (input.get(input.size()-1).getEndAsLocalDateTime().isBefore(startSearchDateTime.plusMinutes((long) dateLenght + 1))) 
+            return null;
```

Comment:
wenn hier der return null ist und unten auch mach doch hier lieber if( ! ...) Schleife durchlaufen, else return null. und dann unten den hinter der schleife raus. also wenn deine bedingung dann wie hier erfüllt ist geht er in den else und returnt null, also gleiche laufzeit aber nur ein return :-)